db_whitebox_test.go

```diff
@@ -112,6 +119,180 @@ func TestMethodPage(t *testing.T) {
 	}
 }
 
+func TestTx_Check_CorruptPage(t *testing.T) {
+	testCases := []struct {
+		name      string
+		bucketKey string
+		isPanic   bool
+	}{
+		{
+			name:      "corrupt page by violating btree invariant",
+			bucketKey: "testBucket",
+			isPanic:   false,
+		},
+		{
+			name:      "corrupt page by dumping random bytes",
+			bucketKey: "testBucket",
+			isPanic:   true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			fileName, err := prepareData(t)
+			require.NoError(t, err)
+
+			t.Logf("Creating db file '%v'.", fileName)
+			db, err := Open(fileName, 0666, &Options{PageSize: 4096})
+			require.NoError(t, err)
+			defer func() {
+				require.NoError(t, db.Close())
+			}()
+
+			uErr := db.Update(func(tx *Tx) error {
+				t.Logf("Creating bucket '%v'.", tc.bucketKey)
+				b, bErr := tx.CreateBucketIfNotExists([]byte(tc.bucketKey))
+				require.NoError(t, bErr)
+				t.Logf("Generating random data in bucket '%v'.", tc.bucketKey)
+				generateSampleDataInBucket(t, b, 3)
+				return nil
+			})
+			require.NoError(t, uErr)
+
+			t.Logf("Corrupting random leaf page in bucket '%v'.", tc.bucketKey)
+			victimPageId := corruptLeafPage(t, db, tc.isPanic)
+
+			if tc.isPanic {
+				defer func() {
+					if r := recover(); r != nil {
+						t.Logf("panic during consistency check: %v.", r)
+					}
+				}()
+			}
+
+			t.Log("Running consistency check.")
+			var cErrs []error
+			vErr := db.View(func(tx *Tx) error {
+				chkConfig := checkConfig{
+					kvStringer: HexKVStringer(),
+				}
+				ch := make(chan error)
+				defer close(ch)
+
+				go func() {
+					for cErr := range ch {
+						cErrs = append(cErrs, cErr)
+					}
+				}()
+
+				tx.check(chkConfig, ch)
```

Comment:
It seems that you only check the corrupted page. Suggest to also check the normal pages (not corrupted pages), and they should be successful.

Reply:
thanks a lot for your review 

but how, `tx.check()` only return errors through the channel, it does not report the successful pages iiuc

Reply:
> but how

We can regard it as successful once the following conditions are all true,
- the tx.check() already returned;
- there is any error
- no panicking

Reply:
done 👍🏽